### PR TITLE
Add go coverprofile format support

### PR DESCRIPTION
**Pull request by bodhi**
Extend Formatter to support both lcov and go coverage data by:

* Implementing basic parser of go coverage format
* Modifying Formatter to inspect coverage file data before deciding on a parser

### CODE REVIEW

File: gocover_parse.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+module.exports = function(data, cb) {
+  var memo = {}
```
Comment (GordonDiggs) [inline gocover_parse.js]: :lipstick: Add a semicolon at the end here

File: gocover_parse.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  var parsedData = []
```
Comment (GordonDiggs) [inline gocover_parse.js]: :lipstick: Add a semicolon at the end here

File: test/test.js [hunk 3]
```diff
@@ -20,6 +22,24 @@ describe('JSON', function(){
...
   });
+
+  describe('gocover', function() {
+    it('should return the correct filenames', function(done) {
+      process.env.GOPATH = rootDirectory; // files will be at $GOPATH/src
+      formatter.format(gocoverFixture, function(err, data) {
+        var names = data.source_files.map(function(elem) {
+          return elem.name;
+        });
+        var expected = [
```
Comment (GordonDiggs) [inline test/test.js]: Is there a way to assert that the coverage is formatted correctly?

**Comment by GordonDiggs** [comment]
@bodhi left you a few small comments (mostly style), but otherwise this looks great! Thank you!

**Comment by bodhi** [comment]
Hi @GordonDiggs I'm in a different timezone (+11), and went to bed straight after I submitted this, so haven't had a chance to fix the semicolons yet. Thanks for merging it!

**Comment by GordonDiggs** [comment]
@bodhi no worries! I merged by accident but will fix up the semicolons myself! I can push this out tomorrow - I'll let you know when it's all done

File: test/test.js (hunk 3 above)
Comment (bodhi) [inline test/test.js]: :+1: (If I make the time/motivation!) I just looked at what subset of the `lcov` output format the formatter used, and tried to generate the same structure.

If another input format was added, I'd consider formalising the data structure format that is passed to CodeClimate (or doubling down on `lcov` output) and splitting the formatter into separate parsing and transformation-to-CodeClimate-API-JSON modules.

**Comment by bodhi** [comment]
> I merged by accident but will fix up the semicolons myself

😁

**Comment by GordonDiggs** [comment]
@bodhi had time now, so this is released to npm! Thanks again

**Comment by bodhi** [comment]
I added it to our Semaphore CI setup, appears to work well. Now we just need to make a change to the test coverage to see the result ;) Thanks for making it open-source so that I could add it, too.

**Comment by devinus** [comment]
This PR appears to have broken thing for some people: https://github.com/codeclimate/javascript-test-reporter/issues/33

**Comment by bodhi** [comment]
Oops, sorry @devinus!

**Pull request merged**